src/mds/MDBalancer.cc

```diff
@@ -102,6 +102,15 @@ void MDBalancer::handle_export_pins(void)
     CInode *in = *cur;
     ceph_assert(in->is_dir());
     mds_rank_t export_pin = in->get_export_pin(false);
+    if (export_pin >= mds->mdsmap->get_max_mds()) {
+      dout(20) << " delay export pin on " << *in << dendl;
+      in->state_clear(CInode::STATE_QUEUEDEXPORTPIN);
```

Comment:
Is there chance that the dir be re-insert into the export_pin_queue with the flag unset?

Reply:
No, there is only one place to insert dir into the queue with this flag, which is maybe_export_pin. Even if in the "re-insert" logic, I still call maybe_export_pin.